### DISCUSSION TITLE
Unable to establish TCP connection for more than 1 device in same network

### DISCUSSION THREAD

**Discussion by nitish098** [openthread/openthread#7797] [Q&A] [answered]
I was able to successfully establish a TCP connection between
an FTD and an MTD. But when I try to establish a TCP connection from this FTD to another MTD, while the other connection exists, the FTD hangs and the TCP connection is lost.

I use
FTD: nRF52840 (ot-cli-ftd),
1st MTD: BRD4166A (ot-cli-mtd),
2nd MTD: nRF52840 (ot-cli-mtd).

Using command :
FTD:
> tcp init
Done
> tcp listen :: 30000
Done

MTD1:
> tcp init
Done
>tcp connect FTD ipaddr 30000
Done
Accepted connection from [fe80:0:0:0:8f3:f602:bf9b:52f2]:49152
TCP: Connection established

MTD2:
> tcp init
Done
>tcp connect FTD ipaddr 30000
Done
Connection failed

Is it due to the TCP connection or the device?

**Comment by jwhui** [comment, marked as answer]
While the OpenThread [TCP API](https://openthread.io/reference/group/api-tcp) is fully capable of supporting multiple TCP connections simultaneously, the current TCP CLI example only supports one TCP end point.

https://github.com/openthread/openthread/blob/171fbd82cf57a399afbbace78514923451ea5da4/src/cli/cli_tcp.hpp#L130
